### PR TITLE
Add default wildcard routing e2e test

### PR DESCRIPTION
**Pull request by ironcladlou**
Add a test which exercises:

* Wildcard DNS management
* Route host defaulting
* Default wildcard certificates

By:

* Creating a reencrypt route using only defaults
* Asserting the route is accessible over its public hostname

This test should only execute when wildcard DNS management is enabled as
indicated by the presence of an ingress domain on the cluster config resource.

### CODE REVIEW

File: test/extended/router/wildcard.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+			o.Expect(err).NotTo(o.HaveOccurred())
+		})
+	})
+})
+
+func waitForURLOK(ns, execPodName, url string, timeoutSeconds int) error {
+	cmd := fmt.Sprintf(`
+		set -e
+		for i in $(seq 1 %d); do
+			code=$( curl -k -s -o /dev/null -w '%%{http_code}\n' %q ) || rc=$?
```
Comment (Miciah) [inline test/extended/router/wildcard.go]: I'd suggest adding `-m 1` to the `curl` command to time out after 1 second.  If you want a higher timeout value, or if you want to observe `timeoutSeconds` more strictly, you could add `if [[ $SECONDS -gt %d ]]; then echo timed out; exit 1; fi` to the loop.

Comment (ironcladlou) [reply test/extended/router/wildcard.go]: Done

**Comment by ironcladlou** [comment]
Okay, finally getting back to this, I made the test only execute if we detect that the default router is fronted by a LoadBalancer service. I'm not sure that's the best way to do detection going forward, but I think it works for now and we desperately need this coverage.

**Comment by ironcladlou** [comment]
https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/21617/pull-ci-openshift-origin-master-e2e-gcp/2714

This should be skipping on GCE so something’s wrong either in my skip detection or in the operator

**Comment by ironcladlou** [comment]
So close!

/retest

File: test/extended/router/wildcard.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+			url := "https://" + hostname
+			g.By(fmt.Sprintf("verifying the route serves 200 from %s", url))
+			err = waitForURLOK(ns, execPodName, url, changeTimeoutSeconds)
+			o.Expect(err).NotTo(o.HaveOccurred())
+		})
+	})
+})
+
+func waitForURLOK(ns, execPodName, url string, timeoutSeconds int) error {
+	cmd := fmt.Sprintf(`
```
Comment (smarterclayton) [inline test/extended/router/wildcard.go]: Is there a reason you didn't reuse the other helpers?

Comment (ironcladlou) [reply test/extended/router/wildcard.go]: All the other ones take/use a Host header, which I don't want in these tests

**Comment by ironcladlou** [comment]
@smarterclayton had to rebase, please re-tag

**Comment by ironcladlou** [comment]
Well, tests passed and instead of merging tests kicked off again for some reason.

/retest

**Comment by ironcladlou** [comment]
What a ride!

**Pull request merged**